XiEditor/CoreConnection.swift

```diff
@@ -66,23 +79,72 @@ class CoreConnection {
         } else {
             self.rpcLogWriter = nil
         }
-        let task = Process()
+        
+        let tmpErrLog = logDirectory.appendingPathComponent("xi_tmp.log").path
+        
+        self.errLogWriter = FileWriter(path: tmpErrLog)
+        if self.errLogWriter != nil {
+            print("logging stderr to \(tmpErrLog)")
+        }
+          
         task.launchPath = path
         task.arguments = []
         let outPipe = Pipe()
         task.standardOutput = outPipe
         let inPipe = Pipe()
         task.standardInput = inPipe
+        let errPipe = Pipe()
+        task.standardError = errPipe
         inHandle = inPipe.fileHandleForWriting
         recvBuf = Data()
-
+        
         outPipe.fileHandleForReading.readabilityHandler = { handle in
             let data = handle.availableData
             self.recvHandler(data)
         }
+        
+        errPipe.fileHandleForReading.readabilityHandler = { handle in
+            let data = handle.availableData
+            var lineCount = 1
+            
+            if let errString = String(data: data, encoding: String.Encoding.utf8) {
+                print(errString, terminator: "")
+                self.errOutput += errString
+            }
+            
+            if self.errOutput.hasSuffix("\n") {
```

Comment:
Okay I get the idea here, but I think it's probably more clever than we need. Since we delete the log if we exit normally we aren't particularly worried about wasting disk space; I would just log everything as it comes in, and not worry about it.

There's also a logic bug here, in that it's possible for errString to include interior (non-trailing) newlines.

Reply:
+1.  Let's just dump directly to disk.  If stderr verbosity is a problem we can either reduce the verbosity or add log rollover.  This also gets rid of the in-memory buffering which means you can lose partial log data if the frontend crashes which is likely to contain the important info you want.